### ISSUE TITLE
Lint fails with referenced schemas declaring arrays of objects

### ISSUE THREAD

**Issue by RossBarnie** [Redocly/redocly-cli#828]
**Describe the bug**
When linting an example of a Response content object with referenced schemas, the command `redocly lint` will raise warnings on the `no-invalid-media-type-examples` rule even if the example is valid. When using directly declared schemas the same warning is not raised.

**To Reproduce**
1. Given this `redocly.yaml` file:
  ```yaml
  apis:
    direct:
      root: direct.json
    referenced:
      root: referenced.json
  styleguide:
    extends: 
      - recommended
    rules:
      no-empty-servers: off
      operation-summary: off
      no-invalid-media-type-examples: error
      operation-4xx-response: off
      operation-operationId: off
      info-license: off
  ```
2. And this OpenAPI file(s)
  `direct.json`
  ```json
  {
    "openapi": "3.0.0",
    "info": {
      "title": "test",
      "description": "foo",
      "version": "1.0.0"
    },
    "paths": {
      "/api/foo": {
        "get": {
          "responses": {
            "200": {
              "description": "car",
              "content": {
                "application/json": {
                  "schema": {
                    "type": "array",
                    "items": {
                      "type": "object",
                      "properties": {
                        "fuel": {
                          "type": "string"
                        },
                        "config": {
                          "type": "object",
                          "properties": {
                            "doors": {
                              "type": "integer"
                            }
                          }
                        }
                      }
                    }
                  },
                  "examples": {
                    "Example 1": {
                      "value": [
                        {
                          "fuel": "petrol",
                          "config": {
                            "doors": 4
                          }
                        }
                      ]
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
  ```
  `referenced.json`
  ```json
  {
    "openapi": "3.0.0",
    "info": {
      "title": "test",
      "description": "foo",
      "version": "1.0.0"
    },
    "paths": {
      "/api/foo": {
        "get": {
          "responses": {
            "200": {
              "description": "car",
              "content": {
                "application/json": {
                  "schema": {
                    "$ref": "#/components/schemas/Cars"
                  },
                  "examples": {
                    "Example 1": {
                      "value": [
                        {
                          "fuel": "petrol",
                          "config": {
                            "doors": 4
                          }
                        }
                      ]
                    }
                  }
                }
              }
            }
          }
        }
      }
    },
    "components": {
      "schemas": {
        "Engine": {
          "type": "object",
          "additionalProperties": false,
          "properties": {
            "fuel": {
              "type": "string"
            }
          }
        },
        "Configuration": {
          "type": "object",
          "additionalProperties": false,
          "properties": {
            "doors": {
              "type": "integer"
            }
          }
        },
        "Car": {
          "allOf": [
            {
              "$ref": "#/components/schemas/Engine"
            },
            {
              "type": "object",
              "properties": {
                "config": {
                  "$ref": "#/components/schemas/Configuration"
                }
              }
            }
          ],
          "example": {
            "fuel": "petrol",
            "config": {
              "doors": 4
            }
          }
        },
        "Cars": {
          "type": "array",
          "items": {
            "$ref": "#/components/schemas/Car"
          }
        }
      }
    }
  }
  ```
3. Run this command with these arguments... `redocly lint *.json`

  ```
  validating /direct.json...
  /direct.json: validated in 24ms
  
  validating /referenced.json...
  [1] referenced.json:24:25 at #/paths/~1api~1foo/get/responses/200/content/application~1json/examples/Example 1/value/0/config
  
  Example value must conform to the schema: `0` property must NOT have additional properties `config`.
  
  22 | {
  23 |   "fuel": "petrol",
  24 |   "config": {
  25 |     "doors": 4
  26 |   }
  
  referenced from referenced.json:15:35
  
  Error was generated by the no-invalid-media-type-examples rule.
  
  
  [2] referenced.json:24:25 at #/paths/~1api~1foo/get/responses/200/content/application~1json/examples/Example 1/value/0/config
  
  Example value must conform to the schema: `0` property must NOT have additional properties `config`.
  
  22 | {
  23 |   "fuel": "petrol",
  24 |   "config": {
  25 |     "doors": 4
  26 |   }
  
  referenced from referenced.json:15:35
  
  Error was generated by the no-invalid-media-type-examples rule.
  
  
  [3] referenced.json:23:25 at #/paths/~1api~1foo/get/responses/200/content/application~1json/examples/Example 1/value/0/fuel
  
  Example value must conform to the schema: `0` property must NOT have additional properties `fuel`.
  
  21 | "value": [
  22 |   {
  23 |     "fuel": "petrol",
  24 |     "config": {
  25 |       "doors": 4

  ```

**Expected behavior**
No errors reported, as the two provided examples and schemas are equivalent.

**Logs**
Shown above

**OpenAPI definition**
As above

**Redocly Version(s)**
1.0.0-beta.107

**`Node.js` Version(s)**
```
$ node --version
v14.17.6
$ npx --version
6.14.15
```
Also used in docker containers using the `18-alpine` tagged image with the same result, ie:
```
docker run -u $(id -u) -v $(pwd):/data -w /data node:18-alpine npx @redocly/cli lint
```

**Additional context**

**Comment by tatomyr** [comment]
Hi @RossBarnie,

Thank you for creating the issue. It must be related to https://github.com/Redocly/redocly-cli/issues/552. 

I'm closing this one as duplicated.

**Comment by SmoliyY** [comment]
Hi @RossBarnie ! This issue fixed in [1.0.0.beta-110](https://redocly.com/docs/cli/changelog/#100-beta110-2022-09-21) version. Please, let us know if it works for you, thx)

**Comment by RossBarnie** [comment]
@SmoliyY The example in the original PR has been partially fixed but there's still an error:

```
$ docker run -u $(id -u) -v $(pwd):/data -w /data node:18-alpine npx @redocly/cli lint
npm WARN exec The following package was not found and will be installed: @redocly/cli@1.0.0-beta.110
validating /direct.json...
/direct.json: validated in 44ms

validating /referenced.json...
[1] referenced.json:24:25 at #/paths/~1api~1foo/get/responses/200/content/application~1json/examples/Example 1/value/0/config

Example value must conform to the schema: `0` property must NOT have additional properties `config`.

22 | {
23 |   "fuel": "petrol",
24 |   "config": {
   |   ^^^^^^^^
25 |     "doors": 4
26 |   }

referenced from referenced.json:15:35 at #/paths/~1api~1foo/get/responses/200/content/application~1json

Error was generated by the no-invalid-media-type-examples rule.


/referenced.json: validated in 19ms

❌ Validation failed with 1 error.
run `openapi lint --generate-ignore-file` to add all problems to the ignore file.
```

In another use case I have known problems that aren't flagged with errors. I'll see if I can recreate a minimal example.
